common/Reducers.jsm

```diff
@@ -457,6 +462,26 @@ function DiscoveryStream(prevState = INITIAL_STATE.DiscoveryStream, action) {
       return {...prevState, lastUpdated: INITIAL_STATE.DiscoveryStream.lastUpdated, layout: INITIAL_STATE.DiscoveryStream.layout};
     case at.DISCOVERY_STREAM_FEEDS_UPDATE:
       return {...prevState, feeds: action.data || prevState.feeds};
+    case at.DISCOVERY_STREAM_SPOCS_ENDPOINT:
+      return {
+        ...prevState,
+        spocs: {
+          spocs_endpoint: action.data || prevState.spocs.spocs_endpoint,
+          data: prevState.spocs.data,
```

Comment:
Also why would you leave the data for the previous spocs if the endpoint changed?

Reply:
Can you describe the behaviour for this in a comment actually since it's kinda complex